webapp/src/ts/effects/contacts.effects.ts

```diff
@@ -106,10 +106,14 @@ export class ContactsEffects {
   }
 
   private loadReports(forms) {
+    const initialContactId = this.selectedContact._id;
     return this.contactViewModelGeneratorService
       .loadReports(this.selectedContact, forms)
       .then(reports => {
-        return this.contactsActions.receiveSelectedContactReports(reports);
+        // Make sure the selected contact did not change while the reports were fetched
+        if(this.selectedContact._id === initialContactId) {
```

Comment:
Loading reports is just one part of this issue. This step is followed by loading the target doc, loading contact summary and tasks. We need to stop all of these processes so we don't pollute the state if the contact changed in the meantime.

Reply:
The best case scenario if this happens right now is that we just load everything twice (because we start the other loading promise chain before the first one succeeds).

Reply:
@dianabarsan yes, that makes sense that we should just short-circuit out of the rest of that promise chain.  One further things I was wondering is if we need similar logic in the places where we are loading children/TargetDoc/Tasks/ContactSummary?  It seems like the selectedConact could change during any of those async calls and then the state would still end up polluted.

Reply:
That's correct, all those actions should to be short-circuited (basically interrupting the promise chain entirely) when the selected contact changes (as the selection itself will trigger another instance of the promise chain).